plugins/inputs/logfile/logfile.go

```diff
@@ -205,6 +216,11 @@ func (t *LogFile) FindLogSrc() []logs.LogSrc {
 				continue
 			}
 
+			if err := t.tailerSemaphore.Acquire(ctx, 1); err != nil {
+				t.Log.Errorf("Failed increase counter in semaphore:", err)
+				continue
+			}
```

Comment:
So this is not the original intended behavior. The original purpose of the PR was just to monitor and count the number of log files being monitored. Now, this introduces a leak, because `tail.TailFile()` has already been called so the file is going to start being monitored, but not propagated to the list of LogSrcs.

Reply:
Same previous comment.
```
I'm still doing the benchmark between semaphore and only pushed it for committing a snapshot. Have not done the final PR yet.
```